src/components/Form/index.js

```diff
@@ -0,0 +1,184 @@
+import React, { Component } from 'react'
+import dot from 'dot-object'
+
+export default class Form extends Component {
+   static propTypes = {
+      children: React.PropTypes.node,
+      onError: React.PropTypes.func,
+      onSubmit: React.PropTypes.func
+   }
+
+   constructor(props) {
+      super(props)
+
+      this.handleSubmit = this.handleSubmit.bind(this)
+
+      this.initialValuesLoaded = false
+
+      // Setup state
+      this.state = {}
+      props.children.forEach((child) => {
```

Comment:
I would suggest having all this data inside another property in the state, to avoid potential key name conflicts with other state properties (or potential state properties). Eg:

```js
this.state = {
   isPristine: true,
   formData: {} // This is where initialValues and all form data goes
}
```